CMakeLists.txt

```diff
@@ -294,6 +294,14 @@ endif()
 
 
 # configuration-specific source files
+
+# we don't check for win32 as on win32 socket operations are handled directly through return values
+if(APPLE) # MacOS, iOS
```

Comment:
Rather than basing these checks on the detected configuration platform, please perform the check using `check_symbol_exists` with `SO_NOSIGPIPE` and `MSG_NOSIGNAL`, and then define wrapper based on the resulting `HAVE_SO_NOSIGPIPE` and `HAVE_MSG_NOSIGNAL` `#define` s. This will allow platforms that supply one or the other of these to build correctly, without relying on CMake to bin them as `APPLE` or `UNIX`.